### ISSUE TITLE
Too many levels of symbolic links for node_modules

### ISSUE THREAD

**Issue by fgrehm** [paketo-buildpacks/yarn-install#406]
_Follow up of https://github.com/paketo-buildpacks/yarn-install/issues/397 & https://github.com/paketo-buildpacks/yarn-install/pull/399_

## Expected Behavior

`node_modules` should be symlinked to the right directory at all times, regardless of cache layers used and other collaborating buildpacks

## Current Behavior

When I run the second build of a node app with cache enabled I noticed that the `node_modules` directory exists but it has a broken symlink.

## Possible Solution

Could this be some sort of regression from https://github.com/paketo-buildpacks/yarn-install/pull/399?

## Steps to Reproduce

See [this new script](https://gist.github.com/fgrehm/51c131406f8d6b3ac56f2647175f8f5c#file-01-run-bash) and [output logs](https://gist.github.com/fgrehm/51c131406f8d6b3ac56f2647175f8f5c#file-02-logs-txt). Notice that the build `#3` [yields a broken `node_modules`](https://gist.github.com/fgrehm/51c131406f8d6b3ac56f2647175f8f5c#file-02-logs-txt-L388-L416) while build `#2` passing in `--clear-cache` to `pack build` works fine again. It's basically the same code from https://github.com/paketo-buildpacks/yarn-install/issues/397 but updated to the new `yarn-install` BP release and new output logs.

TL;DR - `./node_modules` is linked to `/tmp/node_modules`, which in turn is linked to `/layers/paketo-buildpacks_yarn-install/launch-modules/node_modules` and finally that points back to `/tmp/node_modules`, resulting in the error mentioned on the issue title

## Motivations

Same as https://github.com/paketo-buildpacks/yarn-install/issues/397

**Comment by thitch97** [comment]
@fgrehm Thanks for opening this issue!  It seems that caching build layers with `pack`'s `--cache-image` flag causes the buildpack to take a surprising path (at least to me). On rebuild, `build-modules` are reused from cache but `launch-modules` layer metadata is not preserved and thus prevents layer reuse. There is buildpack logic which exacerbates the effects of this behavior, which I'll create a fix PR for shortly.
